lib/crystalball/map_generator/simple_map.rb

```diff
@@ -4,30 +4,57 @@ module Crystalball
   class MapGenerator
     # Basic map object for storing execution maps to storage
     class SimpleMap
-      def initialize(storage)
-        @storage = storage
-        @raw_map = {}
+      extend Forwardable
+
+      # Simple data object for map metadata
+      class Metadata
+        attr_accessor :commit, :type
+
+        def initialize(commit: nil, type: nil)
+          @commit = commit
+          @type = type
+        end
+
+        def to_h
+          {type: type, commit: commit}
+        end
       end
 
-      def load
-        self.raw_map = storage.load
+      attr_reader :cases, :metadata
+
+      delegate %i[commit commit=] => :metadata
+
+      def initialize(storage, metadata: nil, cases: {})
+        @storage = storage
+        @cases = cases
+        @metadata = Metadata.new(metadata || {type: self.class.name})
       end
 
       def stash(case_map)
-        raw_map[case_map.case_uid] = case_map.coverage
+        cases[case_map.case_uid] = case_map.coverage
       end
 
       def dump
-        storage.dump raw_map
+        storage.dump self
       end
 
-      def clear!
-        self.raw_map = {}
+      def reset!
```

Comment:
I don't like that we have `clear!` and `reset!`, both have pretty much the same meaning, and you can't tell the difference without checking source.